Configuration guide for external DB on Strapi Cloud (#1655)
strapi/documentation: docusaurus/docs/cloud/advanced/database.md

---
title: Database
displayed_sidebar: cloudSidebar
description: Configure your own database on Strapi Cloud.
canonicalUrl: https://docs.strapi.io/cloud/advanced/database.html
---

# Database

Strapi Cloud provides a pre-configured PostgreSQL database by default. However, you can also configure it to utilize an external SQL database, if needed.

:::caution

> While supported, it is not recommended to use an external database with Strapi Cloud, unless there is an explicit need to do so. This is because Strapi Cloud provides a managed database that is optimized for Strapi. Using an external database may result in unexpected behavior and/or performance issues (e.g., network latency may impact performance). Strapi cannot provide security or support for external databases.

Reviewer: ```suggestion
While possible, it is not recommended to use an external database with Strapi Cloud, unless there is an explicit need to do so. This is because Strapi Cloud provides a managed database that is optimized for Strapi. Using an external database may result in unexpected behavior and/or performance issues (e.g., network latency may impact performance). Strapi cannot provide security or support with configuring external databases.
```

- If using an existing database, the schema must match the Strapi project schema.

:::

## Configuration

The project `./config/database.js` or `./config/database.ts` file must match the configuration found in the [environment variables in database configurations](https://docs.strapi.io/dev-docs/configurations/database#environment-variables-in-database-configurations) section.

Before pushing changes, add environment variables to the Strapi Cloud project:

1.  Log into Strapi Cloud and click on the corresponding project on the Projects page.
2.  Click on the **Settings** tab and choose **Variables** in the left menu.
3.  Add the following environment variables:

    | Variable                           | Value            |
    | ---------------------------------- | ---------------- |
    | `DATABASE_CLIENT`                  | your_db          |
    | `DATABASE_HOST`                    | your_db_host     |
    | `DATABASE_PORT`                    | your_db_port     |
    | `DATABASE_NAME`                    | your_db_name     |
    | `DATABASE_USERNAME`                | your_db_username |
    | `DATABASE_PASSWORD`                | your_db_password |

>     | `DATABASE_SSL`                     | true             |

Reviewer: Should remove this line, if you specify the nested option this is inherently true

    | `DATABASE_SSL_REJECT_UNAUTHORIZED` | false            |
    | `DATABASE_SCHEMA`                  | public           |

4.  Click **Save**.

:::caution
To ensure a smooth deployment, it is recommended to not change the names of the environment variables.

:::

## Deployment

> To deploy the project and utilize the external database, push the changes from earlier. This will trigger a rebuild and new deployment of the Strapi Cloud project.

Reviewer: This will not trigger a rebuild, the user will need to manually trigger it

Author: A redeploy doesn't automatically rebuild? I did not know that! So they would need to push changes, cancel redeploy (which happens on push), and manually trigger a rebuild / redeploy?

Reviewer: Ah the push itself yes but changing the env vars won't

![deployment](https://res.cloudinary.com/dz7knyfbp/image/upload/v1681936928/deploy_h49uly.png)

Once the application finishes building, the project will use the external database.

## Reverting to the default database

> To revert back to the default database, remove the previously added environment variables related to the external database from the Strapi Cloud project dashboard. This action will trigger a rebuild and new deployment of the Strapi Cloud project.

Reviewer: Same here, will not automatically trigger a rebuild, user will need to do it manually

---
title: Database
displayed_sidebar: cloudSidebar
description: Configure your own database on Strapi Cloud.
canonicalUrl: https://docs.strapi.io/cloud/advanced/database.html
---

# Database

Strapi Cloud provides a pre-configured PostgreSQL database by default. However, you can also configure it to utilize an external SQL database, if needed.

:::caution
While supported, it is not recommended to use an external database with Strapi Cloud, unless there is an explicit need to do so. This is because Strapi Cloud provides a managed database that is optimized for Strapi. Using an external database may result in unexpected behavior and/or performance issues.

### Caveats

- Network latency may impact performance.
- Strapi cannot provide security for external databases.
- Strapi cannot provide support for external databases.

:::

:::prerequisites

- A local Strapi project running on `v4.8.1+`.
- Credentials for external database.

> - If using an existing database,

Reviewer 2: Please join lines 27 and 28 :)

- the schema must match the Strapi project schema.

:::

## Configuration

> It is important to ensure that the project `./config/database.js` or `./config/database.ts` file matches the configuration found in the [Environment variables in database configurations](https://docs.strapi.io/dev-docs/configurations/database#environment-variables-in-database-configurations) section.

Reviewer 2: ```suggestion
The project `./config/database.js` or `./config/database.ts` file must match the configuration found in the [environment variables in database configurations](https://docs.strapi.io/dev-docs/configurations/database#environment-variables-in-database-configurations) section.
```

> :::tip

Reviewer 2: I wouldn't put that in a tip since this seems to be part of the typical procedure.
I suggest rewriting this and the following lines as follows:

> Before pushing changes, add environment variables to the Strapi Cloud project:
>
> 1. Log into Strapi Cloud and click on the corresponding project on the Projects page.
> 2. Click on the **Settings** tab and choose **Variables** in the left menu.
> 3. Add the following environment variables:
> _(add the table from lines 47-57 here — please don't forget to indent the whole table so it appears included in step 3 🙂)_
> 4. Click **Save**.

Before pushing changes, it is recommended to log into the Strapi Cloud project dashboard to add the environment variables.

:::

In the Strapi Cloud project dashboard, navigate to the `Variables` section under the `Settings` tab and click `Add variable`. Once the variables are added, click `Save`.

![environment variables](https://res.cloudinary.com/dz7knyfbp/image/upload/v1681936336/env_var_rw55uy.png)

The following environment variables will need to be set:

| Variable                           | Value            |
| ---------------------------------- | ---------------- |
| `DATABASE_CLIENT`                  | your_db          |
| `DATABASE_HOST`                    | your_db_host     |
| `DATABASE_PORT`                    | your_db_port     |
| `DATABASE_NAME`                    | your_db_name     |
| `DATABASE_USERNAME`                | your_db_username |
| `DATABASE_PASSWORD`                | your_db_password |
| `DATABASE_SSL`                     | true             |
| `DATABASE_SSL_REJECT_UNAUTHORIZED` | false            |
| `DATABASE_SCHEMA`                  | public           |

> :::tip

Reviewer 2: ```suggestion
:::caution
```
I think it should be a caution much more than a tip because not following it could lead to unexpected consequences. Tips are more for alternate ways of doing things and can be skipped, while the reader should not skip caution callouts 🙂
